Library/Homebrew/rubocops/components_order.rb

```diff
@@ -113,26 +113,19 @@ def audit_formula(_node, _class_node, _parent_class_node, body_node)
               next
             end
 
-            if on_macos_blocks.length == 1 && on_linux_blocks.length.zero?
-              problem "you need to define an `on_linux` block within your resource block."
-              next
+            if on_macos_blocks.length == 1
+              on_macos_block = on_macos_blocks.first
+              child_nodes = on_macos_block.body.child_nodes
+              if child_nodes[0].method_name.to_s != "url" && child_nodes[1].method_name.to_s != "sha256"
+                problem "only an url and a sha256 (in the right order) are allowed in a `on_macos` " \
+                        "block within a resource block."
+                next
+              end
             end
 
-            if on_macos_blocks.length.zero? && on_linux_blocks.length == 1
-              problem "you need to define an `on_macos` block within your resource block."
-              next
-            end
+            next unless on_linux_blocks.length == 1
```

Comment:
```suggestion
            next if on_linux_blocks.length != 1
```

but I'm wondering about the `2` or `0` case: it might be worth checking one of those explicitly instead?

Reply:
I reorganised the code to be clearer. I now check the 0, 1 and > 1 cases. The 0 case is sort of a no-op in that loop but let's just use an early next to be clearer.